### PR TITLE
Make running profile clearer and more intuitive to improve usability (#3365)

### PR DESCRIPTION
**Pull request by HappenLee**
This CL mainly made the following modifications:
1. Delete Invalid method in Running Profile Class.
2. Move Memlimit Counter from blockmgr to frgment and add PeakMemUsage Counter
3. Fix the bug of buffer pool memlimit counter
4. Call compute_time_in_profile() before pretty_print() to show the _local_time_percent without child running profile
5. Add TransferThread ThreadToken count in AveThreadToken Counter

ISSUE: #3365

### CODE REVIEW

File: be/src/exec/olap_scan_node.cpp [hunk 1]
```diff
@@ -1038,6 +1038,7 @@ Status OlapScanNode::normalize_binary_predicate(SlotDescriptor* slot, ColumnValu
 
 void OlapScanNode::transfer_thread(RuntimeState* state) {
     // scanner open pushdown to scanThread
+    state->resource_pool()->acquire_thread_token();
```
Comment (imay) [inline be/src/exec/olap_scan_node.cpp:1041]: Can you tell the side effect of this operation? Will this thread hang if there is not enough thread resource?

File: be/src/runtime/runtime_state.cpp [hunk 2]
```diff
@@ -226,8 +226,12 @@ Status RuntimeState::init_mem_trackers(const TUniqueId& query_id) {
 
     // _query_mem_tracker = MemTracker::get_query_mem_tracker(
     //         query_id, bytes_limit, _exec_env->process_mem_tracker());
+
+    auto mem_tracker_counter = ADD_COUNTER(&_profile, "MemoryLimit", TUnit::BYTES);
+    mem_tracker_counter->set(bytes_limit);
+
     _query_mem_tracker.reset(
-            new MemTracker(bytes_limit, runtime_profile()->name(), _exec_env->process_mem_tracker()));
+            new MemTracker(&_profile, bytes_limit, runtime_profile()->name(), _exec_env->process_mem_tracker()));
```
Comment (imay) [inline be/src/runtime/runtime_state.cpp:234]: If you add profile for this memory tracker, why not `_instance_mem_tracker`?

File: be/src/exec/olap_scan_node.cpp (hunk 1 above)
Comment (HappenLee) [reply be/src/exec/olap_scan_node.cpp:1041]: Add this operation just to Make AverageThreadTokens more accurate
   - AverageThreadTokens: 1.00  

Do not warry thread hang on this operation。
acquire_thread_token() just do a lightweight add operation without doing the really thread assignment 

inline void ThreadResourceMgr::ResourcePool::acquire_thread_token() {
    __sync_fetch_and_add(&_num_threads, 1);
}

@imay

File: be/src/runtime/runtime_state.cpp (hunk 2 above)
Comment (HappenLee) [reply be/src/runtime/runtime_state.cpp:234]: yes，I check code。the ```_instance_mem_tracker``` is more appropriate

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**